File: mobile/ios/FH07Mobile/Model/ModelingUiState.swift
import Foundation

struct ModelingUiState: Equatable {
    var nodes: [ModeledNodeUiState] = []
    var lastUpdatedAtElapsedMs: Int64? = nil
    var statusMessage: String = "Waiting for sensor samples to render the modeling scene."
}

struct ModeledNodeUiState: Equatable, Identifiable {
    let sensorId: Int
    var placement: SensorPlacement? = nil
    var isLiveInNetworkStatus: Bool = false
    var seq: Int? = nil
    var pitchDeg: Float? = nil
    var rollDeg: Float? = nil
    var gravityVector: ModeledGravityVector? = nil
    var lastSampleReceivedAtElapsedMs: Int64? = nil

    var id: Int { sensorId }
}

struct ModeledGravityVector: Equatable {
    let x: Float
    let y: Float
    let z: Float
}

File: mobile/android/app/src/main/java/com/example/seniordesignmobileapp/model/ModelingUiState.kt
package com.example.seniordesignmobileapp.model

import com.example.seniordesignmobileapp.analysis.SensorPlacement

data class ModelingUiState(
    val nodes: List<ModeledNodeUiState> = emptyList(),
    val lastUpdatedAtElapsedMs: Long? = null,
    val statusMessage: String = "Waiting for sensor samples to render the modeling scene.",
)

data class ModeledNodeUiState(
    val sensorId: Int,
    val placement: SensorPlacement? = null,
    val isLiveInNetworkStatus: Boolean = false,
    val seq: Int? = null,
    val pitchDeg: Float? = null,
    val rollDeg: Float? = null,
    val gravityVector: ModeledGravityVector? = null,
    val lastSampleReceivedAtElapsedMs: Long? = null,
)

data class ModeledGravityVector(
    val x: Float,
    val y: Float,
    val z: Float,
)
